sensor/kubernetes/upgrade/process.go

```diff
@@ -117,6 +118,16 @@ func (p *process) handleCentralCheckIns() {
 			for _, detail := range status.Convert(err).Details() {
 				if _, isNoUpgradeInProgress := detail.(*central.UpgradeCheckInResponseDetails_NoUpgradeInProgress); isNoUpgradeInProgress {
 					log.Info("Central says there is no upgrade in progress. Exiting loop...")
+					func() {
+						ctx, cancel := context.WithTimeout(context.Background(), deleteWaitTimeout)
+						defer cancel()
+						// We should not use p.ctx() as we cannot rely on doneSig to cancel the context as it is triggered after this call.
+						// We need to trigger the doneSig after this call because it is embedded in the k8s client, meaning if doneSig is triggered,
+						// the k8s client will ignore any call as the context is cancelled.
+						if err := p.deleteUpgraderDeploymentIfNecessary(ctx, true); err != nil {
+							log.Errorf("unable to delete the upgrader deployment: %v", err)
```

Comment:
[nit]: I like failed better than unable because I search for fail in logs when something goes wrong. 
```suggestion
							log.Errorf("Failed to delete the upgrader deployment: %v", err)
```

Reply:
I'm not sure about the reasoning behind this one 🤔  The log is an `Error` log so you should search for `Error` when debugging otherwise if you search for `fail` you could miss a lot of things.